enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -270,4 +275,51 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       throw new EnforcerRuleException("Failed to collect dependency " + ex.getMessage(), ex);
     }
   }
+
+  /**
+   * Returns {code Optional} describing the path from project root to a problematic artifact that
+   * caused {@link ArtifactTransferException}. An empty {@code Optional} if {@code exception} is not
+   * caused by {@link ArtifactTransferException}.
+   */
+  @VisibleForTesting
+  static Optional<String> formatDependencyPathInException(DependencyResolutionException exception) {
```

Comment:
No need for Optional. We should only call this method in the case where thios is the type of the exception